pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
This pr doesn't change previous logic. If reconcile fail, we already set status failed.

Actually this PR is fixing it. If reconcile exits and needDeleteCluster = true, it must be error case and must be reported failed